python/tk_premiere/premiere.py

```diff
@@ -201,6 +213,127 @@ def __init__(self, timeline):
         """
         super(PremiereTimeline, self).__init__(timeline)
 
+    @property
+    def node_id(self):
+        """
+        Return the internal ID of the Premiere Sequence project item.
+
+        :returns: A string.
+        """
+        # The Project itself is not a Premiere ProjectItem but its
+        # root item is.
+        return self._item.projectItem.nodeId
+
+    @property
+    def video_tracks(self):
+        """
+        Iterate over all video tracks within this timeline.
+
+        :yields: :class:`PremiereTrack` instances.
+        """
+        # Iterating over tracks directly sometimes
+        # goes into infinite loops or returns ``None``
+        # results, safer to iterate over the number
+        # of tracks.
+        for i in range(self._item.videoTracks.numTracks):
+            track = self._item.videoTracks[i]
+            yield PremiereTrack(track)
+
+    @property
+    def audio_tracks(self):
+        """
+        Iterate over  all audio tracks within this timeline.
+
+        :yields: :class:`PremiereTrack` instances.
+        """
+        # Iterating over tracks directly sometimes
+        # goes into infinite loops or returns ``None``
+        # results, safer to iterate over the number
+        # of tracks.
+        for i in range(self._item.audioTracks.numTracks):
+            track = self._item.videoTracks[i]
```

Comment:
seems suspicious to grab `videoTracks` here, shouldn't it be `self._item.audioTracks[i]`?

Reply:
good catch, thank you!